Update Magento Modules ReadMe (#31304)
magento/magento2: app/code/Magento/Bundle/README.md

# Magento_Bundle module
Magento_Bundle module introduces new product type in the Magento application named Bundle Product.

This module is designed to extend existing functionality of Magento_Catalog module by adding new product type.

For more information about creating product, see [Creating Bundle Product](https://docs.magento.com/user-guide/catalog/product-create-bundle.html)

## Installation details

> Before disabling or uninstalling this module, note that the following modules depends on this module:

Reviewer: ```suggestion
Before disabling or uninstalling this module, note that the following modules depend on this module:
```

- Magento_Catalog
- Magento_GiftMessage
- Magento_Sales

> For information about module installation in Magento 2, see [Enable or disable modules](https://devdocs.magento.com/guides/v2.3/install-gde/install/cli/install-cli-subcommands-enable.html).

Reviewer: ```suggestion
For information about module enabling or disabling in Magento 2, see [Enable or disable modules](https://devdocs.magento.com/guides/v2.3/install-gde/install/cli/install-cli-subcommands-enable.html).
```

[The Magento dependency injection mechanism](http://devdocs.magento.com/guides/v2.1/extension-dev-guide/depend-inj.html) enables you to override the functionality of the Magento_Bundle module.

### Events

The module dispatches the following events:

 - `catalog_product_option_price_configuration_after` in the `\Magento\Bundle\Block\Catalog\Product\View\Type\Bundle::getJsonConfig()` method. Parameters:
    - `configObj` is a config data object will be returned (`\Magento\Framework\DataObject` class).
- `catalog_product_get_final_price` in the methods `\Magento\Bundle\Pricing\Price\BundleSelectionPrice::getValue()`, `\Magento\Bundle\Model\Product\Price::getSelectionFinalTotalPrice()`, `\Magento\Bundle\Model\Product\Price::getFinalPrice`. 
  
    Parameters:
    - `product` is a product with set final price (`\Magento\Catalog\Model\Product` class).
    - `qty` is a qty of product.
- `prepare_catalog_product_collection_prices` in the `\Magento\Bundle\Pricing\Price\BundleSelectionPrice::getValue()` method. Parameters:

>     - `collection` is bundle collection of selection before count prices of it (`\Magento\Bundle\Model\ResourceModel\Selection\Collection` class).

Reviewer: ```suggestion
    - `collection` is bundle collection of selection before price calculation (`\Magento\Bundle\Model\ResourceModel\Selection\Collection` class).
```

>     - `store_id` is a bundle product store id.

Reviewer: ```suggestion
    - `store_id` is a bundle product store ID.
```

- `catalog_product_prepare_index_select` in the `\Magento\Bundle\Model\ResourceModel\Indexer\Price::prepareBundlePriceByType` method. Parameters:
    - `select` is a select for bundle price by type to DB (`\Magento\Framework\DB\Select` class)

>     - `entity_field` is a entity id column (`\Zend_Db_Expr` class)

Reviewer: ```suggestion
    - `entity_field` is a entity ID column (`\Zend_Db_Expr` class)
```

>     - `website_field` is a website id column (`\Zend_Db_Expr` class)

Reviewer: ```suggestion
    - `website_field` is a website ID column (`\Zend_Db_Expr` class)
```

>     - `store_field` is a default store id column (`\Zend_Db_Expr` class)

Reviewer: ```suggestion
    - `store_field` is a default store ID column (`\Zend_Db_Expr` class)
```

> For information about an event in Magento 2, see [Events and observers](http://devdocs.magento.com/guides/v2.1/extension-dev-guide/events-and-observers.html#events).

Reviewer: ```suggestion
For information about the event system in Magento 2, see [Events and observers](http://devdocs.magento.com/guides/v2.1/extension-dev-guide/events-and-observers.html#events).
```

    - `default`
    - `sales_email_order_creditmemo_renderers`
    - `sales_email_order_invoice_renderers`
    - `sales_email_order_renderers`
    - `sales_email_order_shipment_renderers`
    - `sales_order_creditmemo_renderers`
    - `sales_order_invoice_renderers`
    - `sales_order_item_renderers`
    - `sales_order_print_creditmemo_renderers`
    - `sales_order_print_invoice_renderers`
    - `sales_order_print_renderers`
    - `sales_order_print_shipment_renderers`
    - `sales_order_shipment_renderers`

For more information about a layout in Magento 2, see the [Layout documentation](http://devdocs.magento.com/guides/v2.1/frontend-dev-guide/layouts/layout-overview.html).

### UI components

You can extend a product and widgets updates using the configuration files located in the directories 
- `view/adminhtml/ui_component`:
    - 'bundle_product_listing'
- `view/frontend/ui_component`:
    - 'widget_recently_compared'
    - 'widget_recently_viewed'

For information about a UI component in Magento 2, see [Overview of UI components](http://devdocs.magento.com/guides/v2.1/ui_comp_guide/bk-ui_comps.html).

### Public APIs

`\Magento\Bundle\Api\ProductLinkManagementInterface`:

> - get all children for Bundle product.

Reviewer: Please, consider using a lower-cased version of the "bundle" product everywhere (you use both versions)
```suggestion
- get all children for bundle product.
```

> - add a child to bundle product to specified Bundle option by the product sku or product.

Reviewer: ```suggestion
- add a child to bundle product to specified Bundle option by the product SKU or product.
```

# Magento_Bundle module
Magento_Bundle module introduces new product type in the Magento application named Bundle Product.

This module is designed to extend existing functionality of Magento_Catalog module by adding new product type.

## Installation details

Before disabling or uninstalling this module, note that the following modules depend on this module:

- Magento_BundleGraphQl
- Magento_BundleImportExport
- Magento_Wishlist

For information about a module enabling or disabling in Magento 2, see [Enable or disable modules](https://devdocs.magento.com/guides/v2.4/install-gde/install/cli/install-cli-subcommands-enable.html).

## Structure

> `Pricing/` - the directory that contain solutions for bundle product price.

Reviewer: ```suggestion
`Pricing/` - the directory that contains solutions for the bundle product price.
```

## Extensibility

Extension developers can interact with the Magento_Bundle module. For more information about the Magento extension mechanism, see [Magento plug-ins](http://devdocs.magento.com/guides/v2.4/extension-dev-guide/plugins.html).

[The Magento dependency injection mechanism](http://devdocs.magento.com/guides/v2.4/extension-dev-guide/depend-inj.html) enables you to override the functionality of the Magento_Bundle module.

### Events

The module dispatches the following events:

 - `catalog_product_option_price_configuration_after` in the `\Magento\Bundle\Block\Catalog\Product\View\Type\Bundle::getJsonConfig()` method. Parameters:
    - `configObj` is a config data object will be returned(`\Magento\Framework\DataObject` class).
- `catalog_product_get_final_price` in the methods `\Magento\Bundle\Pricing\Price\BundleSelectionPrice::getValue()`, `\Magento\Bundle\Model\Product\Price::getSelectionFinalTotalPrice()`, `\Magento\Bundle\Model\Product\Price::getFinalPrice`. 
  
    Parameters:

>     - `product` - a product with set final price (`\Magento\Catalog\Model\Product` class).

Reviewer: ```suggestion
    - `product` - a product with the final price set (`\Magento\Catalog\Model\Product` class).
```
